### PR TITLE
CI: Making benchmark errors easier to find

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] xref https://github.com/pandas-dev/pandas/pull/29900#issuecomment-559278405

It takes a while to find the failing line of the benchmarks logs at the moment. This should highlight the failing ones, and also publish the log as an artifact so it can be analyzed easier.

CC: @jbrockmendel

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Looks like the desired error messages did show up, not sure if there are some non-desired ones in there too

**Comment by datapythonista** [comment]
Cleaned it a bit, errors were being duplicated, not sure why.

The log when not failures were found was not being shown. Not sure why I did that, but I think it makes sense that it's shown, so restored it.

**Comment by jbrockmendel** [comment]
Sounds good.  Some unwanted errors might go away after #29906.

**Comment by datapythonista** [comment]
@jbrockmendel this looks good to me now. If you can have a look, I'll remove the failure once you're happy.

**Comment by jbrockmendel** [comment]
nice!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -80,15 +80,18 @@ jobs:
         git fetch upstream
         if git diff upstream/master --name-only | grep -q "^asv_bench/"; then
             asv machine --yes
-            ASV_OUTPUT="$(asv dev)"
-            if [[ $(echo "$ASV_OUTPUT" | grep "failed") ]]; then
-                echo "##vso[task.logissue type=error]Benchmarks run with errors"
-                echo "$ASV_OUTPUT"
+            asv dev | sed "/failed$/ s/^/##[error]/" | tee benchmarks.log
```
Comment (WillAyd) [inline .github/workflows/ci.yml]: Is there a reason for using `tee` here instead of simply redirecting the `sed` output?

**Comment by jreback** [comment]
ok lets try it out

File: .github/workflows/ci.yml (hunk 1 above)
Comment (datapythonista) [reply .github/workflows/ci.yml]: I want the log to be shown in the CI, tee gets it printed (besides saving the file later used to see if there are failures and to save the artifact)

**Pull request merged**